version: "3.8"

services:
  dummy:
    image: python:3.12-alpine
    command: "python -m http.server -d /public/"
    environment:
      - PYTHONUNBUFFERED=1
      - _DEPLOY_BUMP=0
    volumes:
      - "./index.html:/public/index.html:ro"
    deploy:
      replicas: 1
      restart_policy:
        condition: on-failure
      update_config:
        parallelism: 1
        delay: 10s
        order: start-first
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.dummy.middlewares=allow-public@file"
        - "traefik.http.routers.dummy.rule=Host(`dummy.oliverlambson.com`) || HostRegexp(`^dummy(\\..+)?$`)"
        - "traefik.http.routers.dummy.entrypoints=websecure"
        - "traefik.http.routers.dummy.tls.certresolver=letsencrypt"
        - "traefik.http.services.dummy.loadbalancer.server.port=8000"
        - "traefik.http.routers.dummy.tls.domains[0].main=dummy.oliverlambson.com"
        # - "traefik.http.routers.dummy.tls.domains[0].sans=dummy.dev.v2.oliverlambson.com"
    networks:
      - public

networks:
  public:
    external: true
